modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -853,17 +872,23 @@ private RuntimeException handleStartException(Throwable e) {
 
         LOG.debug(errMsg, e);
 
-        lifecycleManager.stopNode();
+        try {
+            stop();
+        } catch (Exception exception) {
+            exception.addSuppressed(exception);
+        }
 
         return new IgniteException(errMsg, e);
     }
 
-    /**
-     * Stops ignite node.
-     */
-    public void stop() {
-        lifecycleManager.stopNode();
-        restAddressReporter.removeReport();
+    /** Stops ignite node. */
+    public void stop() throws Exception {
+        IgniteUtils.closeAll(
+                lifecycleManager::stopNode,
+                restAddressReporter::removeReport,
+                indexBuilder::close,
+                indexBuildController::close
```

Comment:
I don't think that this is how it's supposed to be. Shouldn't "lifecycleManager" close everything? I don't see all the other components in this list, which means that you most likely did something wrong.

Reply:
What do you offer?

Reply:
I propose stopping these two components the same way we stop every other component

Reply:
From the `org.apache.ignite.internal.app.LifecycleManager` documentation it is only about `IgniteComponent`, not about other types of components.

Reply:
It’s not entirely clear to me what the criteria must be for a class to become a `IgniteComponent`, can you explain? 
If the answer is: to use it in `LifecycleManager`, that doesn't seem like a reasonable reason.

Reply:
Fix it.